recipes/lora_finetune_single_device.py

```diff
@@ -570,8 +584,11 @@ def _loss_step(self, batch: Dict[str, torch.Tensor]) -> torch.Tensor:
         mask = batch.get("mask", None)  # shape [b, s, s]
         input_pos = batch.get("input_pos", None)  # shape [b, s]
 
-        # run model
-        logits = self._model(tokens, mask=mask, input_pos=input_pos)
+        if self.enable_activation_offloading:
+            with OffloadActivations():
+                logits = self._model(tokens, mask=mask, input_pos=input_pos)
```

Comment:
no strong opinion, but i wonder if this can be cleaner (remove if/else).

For example, one idea could be:
```
with OffloadActivations(is_enabled=enable_activation_offload):
        logits = self._model(tokens, mask=mask, input_pos=input_pos)
```

Reply:
done